src/Engine.cpp

```diff
@@ -321,15 +321,27 @@ bool Engine::fill_entity_info_(
         info_model_->update(default_info);
         info_model_->update_selected_entity(
             backend::entity_kind_to_QString(backend::EntityKind::INVALID),
-            "No entity selected");
+            "No entity selected",
+            "UNKNOWN_APP");
     }
     else
     {
         EntityInfo entity_info = backend_connection_.get_info(id);
+        std::string app_id = "UNKNOWN_APP";
+        //  if the entity has the property "app_id" defined
+        if (entity_info.contains("app_id"))
+        {
+            // check if the property is a known app_id
+            if (std::find(std::begin(backend::app_id_str), std::end(backend::app_id_str), entity_info["app_id"]))
+            {
+                app_id = entity_info["app_id"];
+            }
```

Comment:
If the value is not found, it returns the iterator to the last position.
```suggestion
            if (std::find(std::begin(backend::app_id_str), std::end(backend::app_id_str), entity_info["app_id"]) != std::end(backend::app_id_str))
            {
                app_id = entity_info["app_id"];
            }
```